lib/Runtime/PlatformAgnostic/Intl.h

```diff
@@ -35,56 +77,60 @@ namespace Intl
         void Cleanup() override
         {
             delete intlObject;
-            intlObject = nullptr;
+            delete this;
```

Comment:
I know we have others but I really don't like `delete this` .. Also -> https://isocpp.org/wiki/faq/freestore-mgmt#delete-this

Reply:
This will be going away very soon as Intl.cpp/h code will be moving back into IntlEngineInterfaceExtensionObject. I am not sure if its a good idea to add new code to the Intl-ICU branch at this point, since the only way that this enormous PR is bearable to review is because most of the code has already been reviewed in earlier PRs. I can add an issue for it?